embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/anr/detection/ResponsivenessMonitor.kt

```diff
@@ -0,0 +1,106 @@
+package io.embrace.android.embracesdk.anr.detection
+
+import com.google.gson.annotations.SerializedName
+import io.embrace.android.embracesdk.internal.clock.Clock
+import java.util.concurrent.atomic.AtomicLong
+
+/**
+ * Enables the monitoring of a component or thread's responsiveness by bucketing the time lapsed between calls of [ping] into the specified
+ * durational buckets separated by the given boundaries. The counts of the buckets are provided along with start and end times of the
+ * first N outliers, defined by any interval greater than [outlierThreshold], where N specified by [outliersLimit].
+ *
+ * The bucket name is the boundary value (inclusive) down to the next boundary value or 0. The last bucket is anything greater than the
+ * last boundary and less than [Long.MAX_VALUE].
+ *
+ * Note that this does not handle backpressure so don't use it if the rate of pings is expected to exceed the rate at which the pings can
+ * be processed sequentially.
+ */
+internal class ResponsivenessMonitor(
```

Comment:
Instances of this is currently only used on a single (background) thread, but I synchronized access out of habit. In the old days, I remember there being a small perf penalty on the JVM to locking even if the code runs on the same thread. Is it the same for ART?

If we're not sure, I can write a mircrobenchmark test to check it out if there's a material difference. I feel like keeping it for now in case this gets used on multiple threads later.

Reply:
I'm reasonably sure there is still a small perf penalty, but I think it makes sense to keep this as-is for now.

Reply:
no free lunch, etc. this is on a the background thread so I care less, but I do want to profile this eventually